wp-includes/rest-api/auth/class-wp-rest-key-pair.php

```diff
@@ -0,0 +1,798 @@
+
+	/**
+	 * Initializes the class.
+	 *
+	 * @since 0.1
+	 *
+	 * @see add_action()
+	 * @see add_filter()
+	 */
+	public function init() {
+		add_action( 'rest_api_init', array( $this, 'register_routes' ), 99 );
+		add_action( 'show_user_profile', array( $this, 'show_user_profile' ) );
+		add_action( 'edit_user_profile', array( $this, 'show_user_profile' ) );
+		add_action( 'after_password_reset', array( $this, 'after_password_reset' ) );
+		add_action( 'profile_update', array( $this, 'profile_update' ) );
+
+		add_filter( 'rest_authentication_require_token', array( $this, 'require_token' ), 10, 3 );
+		add_filter( 'rest_authentication_user', array( $this, 'authenticate' ), 10, 2 );
+		add_filter( 'rest_authentication_token_private_claims', array( $this, 'payload' ), 10, 2 );
+		add_filter( 'rest_authentication_validate_token', array( $this, 'validate_token' ) );
+	}
+
+	/**
+	 * Return the REST URI for the endpoint.
+	 *
+	 * @since 0.1
+	 *
+	 * @static
+	 */
+	public static function get_rest_uri() {
+		return sprintf( '/%s/%s/%s', rest_get_url_prefix(), self::_NAMESPACE_, self::_REST_BASE_ );
+	}
+
+	/**
+	 * Registers the routes for the authentication method.
+	 *
+	 * @since 0.1
+	 *
+	 * @see register_rest_route()
+	 */
+	public function register_routes() {
+		$args = array(
+			array(
+				'methods'  => WP_REST_Server::CREATABLE,
+				'callback' => array(
+					$this,
+					'generate_key_pair',
+				),
+				'args'     => array(
+					'name'    => array(
+						'description'       => esc_html__( 'The name of the key-pair.', 'jwt-auth' ),
+						'type'              => 'string',
+						'required'          => true,
+						'sanitize_callback' => 'sanitize_text_field',
+						'validate_callback' => 'rest_validate_request_arg',
+					),
+					'user_id' => array(
+						'description'       => esc_html__( 'The ID of the user.', 'jwt-auth' ),
+						'type'              => 'integer',
+						'required'          => true,
+						'sanitize_callback' => 'absint',
+						'validate_callback' => 'rest_validate_request_arg',
+					),
+				),
+			),
+			'schema' => array(
+				$this,
+				'get_item_schema',
+			),
+		);
+		register_rest_route( self::_NAMESPACE_, '/' . self::_REST_BASE_ . '/(?P<user_id>[\d]+)', $args );
+
+		$args = array(
+			array(
+				'methods'  => WP_REST_Server::DELETABLE,
+				'callback' => array(
+					$this,
+					'delete_all_key_pairs',
+				),
+				'args'     => array(
+					'user_id' => array(
+						'description'       => esc_html__( 'The ID of the user.', 'jwt-auth' ),
+						'type'              => 'integer',
+						'required'          => true,
+						'sanitize_callback' => 'absint',
+						'validate_callback' => 'rest_validate_request_arg',
+					),
+				),
+			),
+		);
+		register_rest_route( self::_NAMESPACE_, '/' . self::_REST_BASE_ . '/(?P<user_id>[\d]+)/revoke-all', $args );
+
+		$args = array(
+			array(
+				'methods'  => WP_REST_Server::DELETABLE,
+				'callback' => array(
+					$this,
+					'delete_key_pair',
+				),
```

Comment:
Absolute style nit but we try to keep these `callback` arrays to one line in the core controller code, to make it more obvious it's a class method lookup.

Reply:
Second that!